lib/dest/prepare.js

```diff
@@ -5,13 +5,15 @@ var path = require('path');
 var fs = require('graceful-fs');
 var through = require('through2');
 
+var LINK_MODE = require('../constants').LINK_MODE;
+
 function prepareWrite(folderResolver, optResolver) {
   if (!folderResolver) {
     throw new Error('Invalid output folder');
   }
 
   function normalize(file, enc, cb) {
-    var mode = optResolver.resolve('mode', file);
+    var mode = !file.symlink ? optResolver.resolve('mode', file) : LINK_MODE;
```

Comment:
Should this check `file.isSymbolic()` like the other parts of the code?  I'm leaning towards yes but I guess we wouldn't need to change the mode then.

Reply:
Should the documentation for `.dest()` options.mode property be updated to note this?

Reply:
Does this even need to do this since we are **only** working with symbolic links for dest?

Reply:
Not sure I follow, sorry... `file.isSymbolic` only works later on _because_ we set mode here. If the vinyl doesn't have the `symlink` property (added by `src` or `symlink) the option resolver will fall back to using the mode already on the vinyl unless a custom one was explicitly given by the user.

I probably misunderstand your comment?

Reply:
@erikkemperman theoretically, we are only receiving Vinyls with "stats" that match that of a Symbolic File (likely because someone used `resolveSymlinks: false` in `.src()` or forwarded by `.symlink()`) so I don't think we should be changing the mode here.  Maybe we just need to disallow the ability for a user to change the mode when `file.isSymbolic()` and note that in the documentation.  Thoughts?

Reply:
I'll chime in, I agree a user shouldn't be able to change the mode.

Reply:
@phated Ah, now I see your point. Yes, if we let go of the scenarios of users manipulating vinyls, we can assume we only end up here directly from `src` with `resolveSymlinks: false` (meaning the vinyl will have the right stats, including mode) or `symlink` (which forces the mode to be symbolic). Then we can indeed simply test `file.isSymbolic()` and won't need to do anything here in that case.